### PR TITLE
new module purchase_stock_analysis

### PR DESCRIPTION
**Pull request by lepistone**

### CODE REVIEW

**Comment by jgrandguillaume** [comment]
Hi,

Thanks for the contrib, that works like a charm !

Functional+code review :+1: 

Regards,

File: purchase_stock_analysis/README.rst [hunk 1]
```diff
@@ -0,0 +1,62 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Purchase Stock Analysis
```
Comment (pedrobaeza) [inline purchase_stock_analysis/README.rst:5]: Add upper line

File: purchase_stock_analysis/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+{
+    'name': 'Purchase Stock Analysis',
+    'version': '1.0',
```
Comment (pedrobaeza) [inline purchase_stock_analysis/__openerp__.py]: Put 8.0.1.0.0

File: purchase_stock_analysis/model/purchase_stock_analysis.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+    _order = 'date_order desc, price_total desc'
+
+    def init(self, cr):
+        tools.sql.drop_view_if_exists(cr, 'purchase_stock_analysis')
+        cr.execute("""
+            create or replace view purchase_stock_analysis as (
+                WITH currency_rate (currency_id, rate, date_start, date_end)
+                AS (
+                    SELECT r.currency_id, r.rate, r.name AS date_start,
```
Comment (pedrobaeza) [inline purchase_stock_analysis/model/purchase_stock_analysis.py]: Use the same technique as Odoo to split the query in sub-methods easily inheritable that compose the SQL

Comment (lepistone) [inline purchase_stock_analysis/model/purchase_stock_analysis.py]: What do you mean? Anyway, this is a variant of an existing reporting view (Purchase Analysis), so maybe it's a good idea to leave it like that.

Comment (pedrobaeza) [inline purchase_stock_analysis/model/purchase_stock_analysis.py]: I mean this: https://github.com/odoo/odoo/blob/8.0/addons/sale/report/sale_report.py#L128

**Comment by lepistone** [comment]
@pedrobaeza updated as you suggest (let's see if I introduced syntax errors splitting the query :fearful: )

**Comment by pedrobaeza** [comment]
Thanks! :+1: Runbot is saying that it's OK :wink:

**Pull request merged**